### PR TITLE
feat: [DHIS2-14008] use the SelectionBar ui component inside ScopeSelector

### PR DESCRIPTION
**Pull request by simonadomnisoru**

### CODE REVIEW

File: src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js [hunk 1]
```diff
@@ -1,185 +1,77 @@
...
-    handleClick(selectedOu) {
+    handleClick(selectedOu: {id: string, displayName: string, code?: string}) {
         const orgUnitObject = { id: selectedOu.id, name: selectedOu.displayName, code: selectedOu.code };
-        this.props.handleClickOrgUnit(selectedOu.id, orgUnitObject);
-    }
-
-    handleReset() {
-        this.props.handleClickOrgUnit(undefined, null);
+        const { handleClickOrgUnit } = this.props;
+        handleClickOrgUnit && handleClickOrgUnit(selectedOu.id, orgUnitObject);
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js:47]: I've noticed that when you are in an enrollment and trying to select an orgUnit, the window does not automatically close. Is this intended or a bug?

File: src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js [hunk 2]
```diff
@@ -1,185 +1,77 @@
...
-                            previousOrgUnitId={this.props.previousOrgUnitId}
-                        />
-                    </div>
-                </Paper>
-            </div>
+            <SelectorBarItem
+                displayOnly={Boolean(selectedOrgUnitId)}
+                label={i18n.t('Registering unit')}
+                noValueMessage={i18n.t('Choose a registering unit')}
+                value={Boolean(selectedOrgUnitId) && selectedOrgUnit?.name}
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js:58]: The component throws an error here when there is no value - we're passing in a bool, while it expected a string. Does this have something to do with the closing issue mentioned above?

image

File: src/core_modules/capture-core/components/Pages/Enrollment/TopBar.container.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                displayOnly
+            />
+            <SingleLockedSelect
+                ready
+                onClear={() => resetEnrollmentId()}
+                onSelect={id => setEnrollmentId({ enrollmentId: id })}
+                options={enrollmentsAsOptions}
+                selectedValue={enrollmentId}
+                title={i18n.t('Enrollment')}
+            />
```
Comment (eirikhaugstulen) [inline src/core_modules/capture-core/components/Pages/Enrollment/TopBar.container.js:75]: When in the enrollment page and deselecting the enrollment, this LockedSelector is disabled - even though there are enrollments available.

**Review by eirikhaugstulen — COMMENTED** [review]
Hey @simonadomnisoru,
This looks great! Had some small comments, but nothing major 👍
Let me know if something is unclear. Thanks!

**Comment by simonadomnisoru** [comment]
> I've made a PR with some design changes: #3169 (it has a single failing test, but I can't see how it's related to the changes I made).
> 
> It would be good to make some changes to the org. unit selector component (it seems slightly out of sync with the UI component), but I think that's out of scope for this PR.

Hey @cooper-joe, 

The design changes are merged. Don't worry about the cypress test falling, I'll take a look at it.
I agree that some UI improvements need to be made to the org unit selector (@karolinelien also suggested to gray-out the unavailable org units). But as you mentioned, I think this is out of this PR's scope and we can address it in a separate issue.

Thank you for the feedback!

**Comment by jasminenguyennn** [comment]
hey @simonadomnisoru. Well done, the PR is great 🎉
few small cosmetic comments:
- when clicking inside the org unit selector, I noticed the position of the dropdown jump after every click 

https://user-images.githubusercontent.com/89806888/212922302-13f78a80-23c2-4d16-9eef-9b2a78111a1d.mov


- not sure if it's by design but when open the program selector, the input field is autofocus
Screenshot 2023-01-17 at 14 18 10

File: src/core_modules/capture-core/components/Pages/Enrollment/TopBar.container.js (hunk 3 above)
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/Pages/Enrollment/TopBar.container.js]: Good catch. I updated the code to fix this issue as well as the rest of your comments. Thanks!

**Comment by simonadomnisoru** [comment]
> hey @simonadomnisoru. Well done, the PR is great 🎉 few small cosmetic comments:
> 
> * when clicking inside the org unit selector, I noticed the position of the dropdown jump after every click
> * not sure if it's by design but when open the program selector, the input field is autofocus
> 

Hey @jasminenguyennn,

I fixed the org unit jumps in the last commit. Regarding the program selector search field autofocus, this is the intended behavior that UX wanted to have.

Thank you for the review!

File: src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js [hunk 4]
```diff
@@ -1,185 +1,79 @@
...
-                        <OrgUnitField
-                            data-test="org-unit-field"
-                            onSelectClick={this.handleClick}
-                            previousOrgUnitId={this.props.previousOrgUnitId}
-                        />
-                    </div>
-                </Paper>
-            </div>
+            <SelectorBarItem
+                displayOnly={Boolean(selectedOrgUnitId)}
```
Comment (cooper-joe) [inline src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js:55]: I tried to remove this line in [my PR](https://github.com/dhis2/capture-app/pull/3169/files#diff-4a059b78a959a9bdf0774576b0a07733742d209f1252ba4a2e3a8609f56094b4L55), but it doesn't seem to have taken effect here. After testing this, the org. unit selector shouldn't be `displayOnly` in any case, I don't think.

Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js]: I must have added it by error when I fixed the merge conflicts 😅 . The last commit fixes it. Thanks!

Comment (cooper-joe) [reply src/core_modules/capture-core/components/ScopeSelector/QuickSelector/OrgUnitSelector.component.js]: Thanks 👍

**Comment by cooper-joe** [comment]
Looking very good! I think there's only one thing left that's not quite right:

- When clearing the organisation unit selection from the `X` icon in the `SelectorBarItem`, the org. unit remains "selected" in the menu. I would expect that the selection inside the menu is also cleared. Here's a screen grab of the issue:


https://user-images.githubusercontent.com/33054985/214504400-87429744-4068-4a8c-9923-824a963f8f03.mp4

**Comment by cooper-joe** [comment]
> This is the intended behavior. After the org unit is deselected the tree expands and highlights the previously selected one. This feature was implemented 6 months ago in [DHIS2-13472](https://dhis2.atlassian.net/browse/DHIS2-13472).

I see, I thought this feature only covered expanding the tree, but not highlighting the previously selected? Is it possible to expand, but not highlight?

**Comment by simonadomnisoru** [comment]
Hey @jasminenguyennn and @eirikhaugstulen,
All comments have been addressed and the PR is ready for review. Please take a look when you have some time.
Thank you! 🙏

**Review by eirikhaugstulen — APPROVED** [review]
Looks great @simonadomnisoru - well done!

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.40,2.39.2,2.38.4 versions

**Pull request merged**